site/docs/components/password_input.mdx

```diff
@@ -0,0 +1,97 @@
+---
+name: Password input
+menu: Components
+route: /components/password-input
+---
+
+import { Playground } from "docz";
+import { StatusLabel } from "hds-react";
+
+import ColorBox from "../../src/components/ColorBox";
+import LargeParagraph from "../../src/components/LargeParagraph";
+import Link from "../../src/components/Link";
+import Text from "../../src/components/Text";
+
+# Password input
+
+<StatusLabel type="alert">Draft</StatusLabel>
+<StatusLabel type="success" style={{marginLeft: 'var(--spacing-xs)'}}>Accessible</StatusLabel>
+
+<LargeParagraph>
+    A password field is a foundational input field component that the user can interact with and specifically input password data.
+</LargeParagraph>
+
+## Principles
+- A label should always be provided with password fields.
+- If possible, allow the password input to reflect the strength of the password entered.
+- If possible, add other programmatic assistance. Detect and suggest changes to provide the user with assistance in generating strong passwords.
+- If possible, do input validation in client-side real-time and provide the user with immediate feedback. See HDS Form validation pattern page for more information and best practices with password fields.
+- Password fields should always be mandatory and it is recommended to mark it as a required field.
+- Use `autocomplete="new-password"` and `id="new-password"` for the password input in a sign-up form, and for the new password in a reset-password form.
+- Provide functionality to mask/unmask the password.
+- Use `aria-label` and `aria-describedby` for password inputs.
+- Don't double-up; Instead, it's better to enable users to confirm their email addresses.
+
+## Accessibility
+- Use element attributes to access browser features: type, name, autocomplete, required, autofocus.
+- Place sign-in in its own `form` element.
+- Ensure the successful form submission
+
+## Security
+As HDS password input provides only the front-end solution, remember that you need to take care of the security in your project.
+Here are some good practices you need to consider.
+
+### General Security
```

Comment:
Tässä osiossa rupeaa olemaan aika paljon tavaraa. Mietin, että olisiko järkevää jo kohta luoda tälle sisällölle oma sivunsa?

Reply:
Voisi olla sekin. Toistaiseksi tää salasanakenttä on ainoa selkeästi tietoturvakysymys, toisaalta esim. lomakkeiden käsittelyssä noin kaiken kaikkiaan pitäisi näitä huomioida, joten voisi ehkä kirjoittaa erilliselle sivulle laajemmin, ja linkittää tässä.

Reply:
Nämä on muutenkin hieman enemmän dev.hel.fi -asiaa, eli sitäkin sijaintia voisi miettiä. Toisaalta täältä ne varmaan löytyvät helpommin kun joku näitä ottaa käyttöön.